thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/auth/ThirdEyeAuthorizer.java

```diff
@@ -37,6 +38,14 @@ boolean authorize(
   // if the implementation does not support multiple namespace, please return a list with the value null Collections.singletonList(null)
   @NonNull List<String> listNamespaces(final ThirdEyePrincipal principal);
 
+  /**
+   * Generates the datasource connection configuration given principal
+   *
+   * @param principal the principal to generate datasource for
+   * @return DataSourceApi
+   */
+  DataSourceApi generateDatasourceConnection(final ThirdEyePrincipal principal);
```

Comment:
let's add `@Nullable`to the return value and explain the meaning of null in the doc. 

smthing like
```
Returns null if there is no recommended datasource configuration. 
```